vllmd-hypervisor/generate-runtime-image.sh

```diff
@@ -0,0 +1,593 @@
+        echo "Checking for existing macvtap devices..."
+        if ip link show macvtap0 &>/dev/null; then
+            echo "Macvtap device macvtap0 already exists, removing it first..."
+            sudo ip link set macvtap0 down || { 
+                echo "ERROR: Failed to bring down existing macvtap device"; 
+                echo "Please manually remove it with: sudo ip link set macvtap0 down"; 
+                exit 1;
+            }
+            
+            sudo ip link delete macvtap0 || {
+                echo "ERROR: Failed to delete existing macvtap device";
+                echo "Please manually remove it with: sudo ip link delete macvtap0";
+                exit 1;
+            }
+        fi
+        
+        # Get the default network interface
+        local default_interface=$(ip route | grep default | head -1 | awk '{print $5}')
+        if [[ -z "${default_interface}" ]]; then
+            echo "ERROR: Could not determine default network interface"
+            exit 1
+        fi
+        echo "Host network interface is ${default_interface}"
+        
+        # Define macvtap device name and MAC address
+        local macvtap_device="macvtap0"
+        local mac_address="52:54:00:12:34:56"
+        
+        # Create a macvtap on the host network
+        echo "Creating macvtap device ${macvtap_device} on ${default_interface}..."
+        sudo ip link add link "${default_interface}" name "${macvtap_device}" type macvtap || {
+            echo "ERROR: Failed to create macvtap device";
+            exit 1;
+        }
+        
+        # Set MAC address and bring up the interface
+        echo "Configuring macvtap device with MAC ${mac_address}..."
+        sudo ip link set "${macvtap_device}" address "${mac_address}" up || {
+            echo "ERROR: Failed to configure macvtap device";
+            sudo ip link delete "${macvtap_device}" || true;
+            exit 1;
+        }
+        
+        # Show macvtap info
+        echo "Macvtap device information:"
+        sudo ip link show "${macvtap_device}"
+        
+        # Get the tap index for the device file
+        echo "Getting tap device index..."
+        local tap_index=$(cat /sys/class/net/"${macvtap_device}"/ifindex 2>/dev/null)
+        if [[ -z "${tap_index}" ]]; then
+            echo "ERROR: Failed to get tap index for ${macvtap_device}"
+            sudo ip link delete "${macvtap_device}" || true
+            exit 1
+        fi
+        
+        # Create the tap device path
+        local tap_device="/dev/tap${tap_index}"
+        echo "Using tap device: ${tap_device}"
+        
+        # Ensure the tap device file exists and is accessible
+        if [[ ! -e "${tap_device}" ]]; then
+            echo "ERROR: Tap device file ${tap_device} does not exist"
+            sudo ip link delete "${macvtap_device}" || true
+            exit 1
+        fi
+        
+        # Make the tap device file accessible
+        sudo chmod 666 "${tap_device}" || {
```

Comment:
make this with a TODO(SECURITY) and indicate we shouldn't modify permissions in this way For now, we will move forward.